encoding/graph/thing/edge.rs

```diff
@@ -614,100 +592,76 @@ impl<'a> ThingEdgeRolePlayerIndex<'a> {
     const LENGTH: usize = PrefixID::LENGTH + 3 * ObjectVertex::LENGTH + 2 * TypeID::LENGTH;
     pub const LENGTH_PREFIX_FROM: usize = PrefixID::LENGTH + ObjectVertex::LENGTH;
 
-    pub fn new(bytes: Bytes<'a, BUFFER_KEY_INLINE>) -> Self {
-        let index = ThingEdgeRolePlayerIndex { bytes };
-        debug_assert_eq!(index.prefix(), Self::PREFIX);
-        index
+    pub fn new(bytes: Bytes<BUFFER_KEY_INLINE>) -> Self {
+        debug_assert_eq!(bytes[Self::INDEX_PREFIX], Self::PREFIX.prefix_id().byte);
+        let player_from = ObjectVertex::new(&bytes[Self::RANGE_FROM]);
+        let player_to = ObjectVertex::new(&bytes[Self::RANGE_TO]);
+        let relation = ObjectVertex::new(&bytes[Self::RANGE_RELATION]);
+        let role_id_from = TypeID::new(bytes[Self::RANGE_FROM_ROLE_TYPE_ID].try_into().unwrap());
+        let role_id_to = TypeID::new(bytes[Self::RANGE_TO_ROLE_TYPE_ID].try_into().unwrap());
+        Self { player_from, player_to, relation, role_id_from, role_id_to }
     }
 
     pub fn build(
-        from: ObjectVertex<'_>,
-        to: ObjectVertex<'_>,
-        relation: ObjectVertex<'_>,
-        from_role_type_id: TypeID,
-        to_role_type_id: TypeID,
-    ) -> ThingEdgeRolePlayerIndex<'static> {
-        let mut bytes = ByteArray::zeros(Self::LENGTH);
-        bytes[Self::RANGE_PREFIX].copy_from_slice(&Self::PREFIX.prefix_id().bytes());
-        bytes[Self::RANGE_FROM].copy_from_slice(from.bytes().bytes());
-        bytes[Self::RANGE_TO].copy_from_slice(to.bytes().bytes());
-        bytes[Self::RANGE_RELATION].copy_from_slice(relation.bytes().bytes());
-        bytes[Self::RANGE_FROM_ROLE_TYPE_ID].copy_from_slice(&from_role_type_id.bytes());
-        bytes[Self::RANGE_TO_ROLE_TYPE_ID].copy_from_slice(&to_role_type_id.bytes());
-        ThingEdgeRolePlayerIndex { bytes: Bytes::Array(bytes) }
-    }
-
-    pub fn prefix_from(
-        from: ObjectVertex<'_>,
-    ) -> StorageKey<'static, { ThingEdgeRolePlayerIndex::LENGTH_PREFIX_FROM }> {
+        player_from: ObjectVertex,
+        player_to: ObjectVertex,
+        relation: ObjectVertex,
+        role_id_from: TypeID,
+        role_id_to: TypeID,
+    ) -> Self {
+        Self { player_from, player_to, relation, role_id_from, role_id_to }
+    }
```

Comment:
This may or may not be a lot of work, though is just renaming - we should flip `new` and `build` now, since `new()` should now be the one that just wraps the struct for the input, while `build()` should be the one that extracts/constructs complex things before returning the struct